src/target/stm32f1.c

```diff
@@ -452,61 +467,59 @@ static bool stm32f1_mass_erase(target *t)
 	if (stm32f1_flash_unlock(t, 0))
 		return false;
 
-	uint32_t sr;
+	bool err;
 	platform_timeout timeout;
 	platform_timeout_set(&timeout, 500);
 
+	stm32f1_flash_clear_eop(t, 0);
+
 	/* Flash mass erase start instruction */
 	target_mem_write32(t, FLASH_CR, FLASH_CR_MER);
 	target_mem_write32(t, FLASH_CR, FLASH_CR_STRT | FLASH_CR_MER);
 
 	/* Wait for completion or an error */
-	do {
-		sr = target_mem_read32(t, FLASH_SR);
-		if ((sr & SR_ERROR_MASK) || !(sr & SR_EOP) || target_check_error(t)) {
-			DEBUG_WARN("stm32f1 flash error 0x%" PRIx32 "\n", sr);
-			return false;
-		}
-		target_print_progress(&timeout);
-	} while (sr & FLASH_SR_BSY);
+	err = stm32f1_flash_busy_wait(t, 0, &timeout);
```

Comment:
Rather than doing this pattern, the following would be preferred and eliminates `bool err` above:
```c
if (!stm32f1_flash_busy_wait(t, 0, &timeout))
    return false;
```